docs/components/InputFile/InputFile.stories.mdx

```diff
@@ -58,3 +58,30 @@ updating an array of `FileUpload`s based on their `key` as uploads progress.
  */
 export function updateFiles(updatedFile: FileUpload, files: FileUpload[]);
 ```
+
+### validator
+
+InputFile can accept a `validator` function as a prop. This will allow you to
+reject or accept file uploads based on your needs. The value of the function
+must accept a File and return `null` when it is accepted or an error object when
+rejected.
+
```

Comment:
tempted to mention this behavior or interaction with `allowedTypes`

![image](https://github.com/GetJobber/atlantis/assets/11843215/a6feb3f1-9027-46ed-af11-a88cb9ad751a)

I don't have a `validator` passed in, but I get the error message showing due to the `options.allow` only being set to images, and I've bypassed that with the sneaky little "show all options" in the native picker dialog

though I suppose these are slightly different?

the `allowedTypes` will attempt to restrict what you can select in that native dialog, and if you skirt it - it will show a rejection message based on that

VS the `validator` won't do anything to restrict what you can pick, it will only tell you after you've submitted if it was invalid with a custom message

that does mean you can end up with something like this though

![image](https://github.com/GetJobber/atlantis/assets/11843215/0f99daf1-3355-494c-8627-fbb0ea451cbc)

Reply:
you know what, let's just call it out anyway

>Note that restricting accepted file types with `allowedTypes` will provide a default error message if an invalid file is selected.

And then for the example below, let's remove the PNG validation since it's a bit confusing to have multiple ways to validate a PNG. let's keep the example limited to validating on attributes other than the file type which you should probably be doing thru `allowedTypes`

Reply:
that makes sense to me to me. I will update the example to just check for file size.